tests/virt_autotest/libvirt_extend_storage_lvm.pm

```diff
@@ -0,0 +1,143 @@
+# SUSE's openQA tests
+#
+# Copyright 2022 SUSE LLC
+# SPDX-License-Identifier: FSFAP
+#
+# Summary: Extend LVM Storage pool / volume test
+#
+# Test flow:
+# - Check a additional free hard disk
+# - Wipe hard disk clean
+# - Partition test disk
+# - Create a pv and display result
+# - Create a vg named 'lvm_vg' and display result
+# - Create a storage pool (lv) named 'guest_image_lvm'
+# - Create a logical volume (lv) size 1G
+# - Attach a logical volume (lv) to guest systems
+# - Clone a logical volume (lv)
+# - Cleanup
+# Maintainer: Leon Guo <xguo@suse.com>
+
+use base "virt_feature_test_base";
+use virt_autotest::utils;
+use virt_autotest::common;
+use strict;
+use warnings;
+use testapi;
+use utils;
+use virt_utils;
+use version_utils 'is_sle';
+
+sub run_test {
+    my ($self) = @_;
+
+    record_info "Prepare Guest Systems";
+    foreach (keys %virt_autotest::common::guests) {
+        start_guests() unless is_guest_online($_);
+    }
+
+    ## About LVM volumes management
+    my ($sys_disk, $dev, $lvm_disk);
+    # Get current system disk name
+    $sys_disk = script_output 'lsblk -rnoPKNAME $(findmnt -nrvoSOURCE /)';
+    # Check with all existed Hard disks
+    my @disks = split(/\n/, script_output("lsblk -n -l -o NAME -d -e 7,11"));
```

Comment:
What does test do if there is only one physical disk in the sut?

Reply:
regarding to lvm storage pool test, requires at least 2 hard drives on our osd sut. 
if there is only one physical disk in the sut, should call return to stop lvm storage pool test. 
currently, depend on the currently VR test results, there are 3-4 hard disk on our osd sut.
So, skip detect only one physical disk situation during perform lvm storage pool test.

Reply:
Not only the SUT's in OSD should be considered, we have also some machines in Beijing and Prague. So I think checking if there are multiple disks is required.

Meanwhile, could you add an extra check for the existing KEEP_DISKS=1? This setting is to reserve other disks (only operate on the disk where the SLE is located). Regard KEEP_DISKS=1 the same way as the machine has one disk only. In some cases, we installed multiple SLE's in other disks but use it as an openqa sut with the setting set.

Reply:
According to current VR, just only focus on our OSD test machine - ipmi_64 backend, missed some machines in Beijing and Prague. 
yes, try to add a check about required hard disk number for lvm storage test.

Reply:
Added a verification of the number of physical hard disks to ensure that there are enough physical hard disks to meet the Virtualization LVM storage test requirements

Reply:
Don't forget to handle KEEP_DISKS=1. we need the test behave the way the machine has only one disk.

Reply:
yes, good point, make sense now. thank you so much for your great help. 
Handle KEEP_DISKS=1 this situation during lvm storage test.